pkg/monitortests/kubeapiserver/faultyloadbalancer/monitortest.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	MonitorName = "faulty-load-balancer"
+)
+
+var (
+	lbTypes = map[string]struct{}{
+		"internal-lb":     {},
+		"external-lb":     {},
+		"service-network": {},
+	}
+)
+
+func NewMonitorTest() monitortestframework.MonitorTest {
+	return &monitorTest{}
+}
+
+type monitorTest struct {
+	monitor *faultyLBMonitor
+}
+
+func (test *monitorTest) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	test.monitor = &faultyLBMonitor{}
+	framework.Logf("monitor[%s]: initialized", MonitorName)
+	return nil
+}
+
+func (test *monitorTest) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	return monitorapi.Intervals{}, nil, nil
+}
+
+func (*monitorTest) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	return nil, nil
+}
+
+func (test *monitorTest) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	intervals := test.monitor.Filter(finalIntervals)
+	if len(intervals) == 0 {
+		return nil, nil
+	}
+	framework.Logf("monitor[%s]: found %d interesting intervals ", MonitorName, len(intervals))
+
+	callback := &junitCallback{name: "[sig-apimachinery] conections (both reused and new) to the kube-apiserver during the graceful termination process should be handled gracefully"}
+	test.monitor.Check(intervals, callback.Evaluate)
+	return callback.Result(), nil
+}
+
+func (*monitorTest) WriteContentToStorage(ctx context.Context, storageDir, timeSuffix string, finalIntervals monitorapi.Intervals, finalResourceState monitorapi.ResourcesMap) error {
+	return nil
+}
+
+func (*monitorTest) Cleanup(ctx context.Context) error {
+	return nil
+}
+
+type junitCallback struct {
+	name     string
+	testCase *junitapi.JUnitTestCase
+}
+
+// constructs the junit test case by taking into account each overlap of
+// a kube-apiserver shutdown interval and an APIUnreachableFromClient interval
+func (junit *junitCallback) Evaluate(shutdown, unreachable monitorapi.Interval) {
+	if junit.testCase == nil {
+		junit.testCase = &junitapi.JUnitTestCase{
+			Name:          junit.name,
+			SystemOut:     fmt.Sprintf("faulty load balancer detected"),
+			FailureOutput: &junitapi.FailureOutput{},
+		}
+	}
+
+	lbType := unreachable.Condition.Locator.Keys[monitorapi.LocatorAPIUnreachableHostKey]
+	msg := fmt.Sprintf("client observed connection error, type: %s\nkube-apiserver: %s\n, client: %s\n", lbType, shutdown.String(), unreachable.String())
+	junit.testCase.FailureOutput.Output = fmt.Sprintf("%s\n%s", junit.testCase.FailureOutput.Output, msg)
+}
```

Comment:
I think this test has to go out in flake only mode, otherwise there's a very good chance it shuts down all payloads. You'll need to refactor a little to return an additional junit testcase with no failure output to trigger it as a flake. 

Sippy can then be used to find occurrences where it flaked. Once you know it's fully passing we can remove that.

Reply:
yes, it makes sense,  it already detects faulty load balancer with AWS,  I have added a new junit testcase with no failure output.